test/python/rest/submit_tests.py

```diff
@@ -0,0 +1,212 @@
+
+        for job in jobs:
+            self.assertTrue(result['name']==job.name, 'submit_job() failed, could not find job')
+            self.assertTrue(job.status=='running', 'submit_job() failed, job is not running, status is ' + job.status + ', health is ' + job.health)
+            self.assertTrue(job.health=='healthy', 'submit_job() failed, job is not healthy, status is ' + job.status + ', health is ' + job.health)
+
+        for job in jobs:
+            logger.warning('store logs for job ' + job.name)
+            filename = job.get_application_logs()
+            self.assertTrue(os.path.isfile(filename), 'get_application_logs() failed, no file stored as ' + filename)
+            #filename = job.store_logs()
+            #self.assertTrue(os.path.isfile(filename), 'store_logs() failed, no file stored as ' + filename)
+            #for pe in job.get_pes():
+            #    logger.warning('store logs for PE ' + pe.id + ' of job ' + job.name)
+            #    filename = pe.store_console_log()
+            #    self.assertTrue(os.path.isfile(filename), 'store_console_log() failed, no file stored as ' + filename)
+            #    filename = pe.store_application_trace()
+            #    self.assertTrue(os.path.isfile(filename), 'store_application_trace() failed, no file stored as ' + filename)
+
+        for job in jobs:
+            logger.warning('cancel job ' + job.name)
+            result = job.cancel()
+            self.assertTrue(result, 'cancel() failed for job ' + job.name)
+
+
+    def test_B_submit_fully_fused(self):
+
+        jobConfig = streamsx.topology.context.JobConfig(target_pe_count=1, tracing='info')
+
+        jobConfigOverlay = {}
+        jobConfig._add_overlays(jobConfigOverlay)
+
+        logger.warning('submit bundle ' + os.path.basename(applicationBundle))
+        result = self.service.submit_job(applicationBundle, configuration=jobConfigOverlay)
+        self.assertFalse('status_code' in result, 'submit_job() failed, status code and error description:' + str(result))
+        self.assertTrue('name' in result, 'submit_job() failed, no job name returned, result: ' + str(result))
+
+        logger.warning('let the job run a while')
+        time.sleep(runInterval)
+
+        logger.warning('check job ' + result['name'])
+        jobs = self.instance.get_jobs(result['name'])
+        self.assertTrue(len(jobs)>0, 'submit_job() failed, could not find job')
+        for job in jobs:
+            self.assertTrue(result['name']==job.name, 'submit_job() failed, could not find job')
+            self.assertTrue(job.status=='running', 'submit_job() failed, job is not running, status is ' + job.status + ', health is ' + job.health)
+            self.assertTrue(job.health=='healthy', 'submit_job() failed, job is not healthy, status is ' + job.status + ', health is ' + job.health)
+            pes = job.get_pes()
+            self.assertTrue(len(pes)==1, 'job configuration failed, job has more than one PE')
```

Comment:
I don't think it's guaranteed that a single PE is created is it?

Reply:
Since the target_pe_count is set in the JobConfig, the fusion_scheme does get set to `manual` in the JobConfig.

Reply:
I think the hint is in the name "target" - not sure it is guaranteed that there will be one pe, though maybe it is if there are no exloaction constraints. Just wondering about intermittent test failures if the scheduler decides it can't fuse into a single pe.

Reply:
This concerns me. Are there situations where Streams will not fuse an application into a single PE when 'fusion_scheme' is 'manual' and 'pe_count' is '1'? Several of our telco customers have deployed applications that depend upon this to meet their performance requirements.